guidance/_program.py

```diff
@@ -306,9 +324,11 @@ def _stream_run(self, loop, new_program):
         yield new_program
 
         # cancel all tasks and close the loop
-        for task in asyncio.all_tasks(loop=loop):
+        for task in self._tasks:
             task.cancel()
         loop.run_until_complete(asyncio.sleep(0)) # give the loop a chance to cancel the tasks
+
+        # TODO: do we really want to close the loop? what if it is used by others?
```

Comment:
@slundberg is it possible that the `loop` is shared with other usages and that this `loop.close()` should not be called (and be the caller's resposability to do that?)

Reply:
Great point.